draft-kasselman-oauth_spiffe.md

```diff
@@ -95,9 +95,9 @@ The SPIFFE deployment is responsible for bootstrapping the identifiers and crede
 
-The OAuth authorization server avoids operational overheads of registering `client_id` and `client_secret` values by accepting SPIFFE credentials from a workload acting as an OAuth client participating in an OAuth flow. The credentials must be signed by a trusted issuer. If this is the first time the authorization server verifies a credential wth a specific SPIFFE ID, it registers a new `client_id`, removing the need for an out-of-band registration flow. The OAuth authorization server may derive a unique `client_id` from the SPIFFE ID, or it may use the SPIFFE ID as the `client_id`. Since the workload is already in possession of a credential that is cryptographically bound to its identifier, no addditional `client_secret` is needed, removing the operational overhead and securtiy risks of managing long-lived secrets in large-scale deployments that already have SPIFFE credentials. The OAuth authorization server may register additional metadata if needed. The authorization server may use a number of mechanissms for obtaining additional metadata. Metadata may be preconfigured, automatically created, retrieved from a configuration management system, or retrieved from the JWT-SVID, if it was included as additional claims, similar to how Dynamic Client Registration ({{RFC7591}}) describes how a software statement may be used to convey metadata.
+The OAuth authorization server avoids operational overheads of registering the client itself and any number of client secrets by accepting SPIFFE credentials from a workload acting as an OAuth client participating in an OAuth flow. The credentials must be signed by a trusted issuer. If this is the first time the authorization server verifies a credential with a specific SPIFFE ID, it registers a new client, removing the need for an out-of-band registration flow. The OAuth authorization server may derive a unique client ID from the SPIFFE ID, or it may use the SPIFFE ID as the client ID. Since the workload is already in possession of a credential that is cryptographically bound to its identifier, no addditional client secret is needed, removing the operational overhead and security risks of managing long-lived secrets in large-scale deployments that already have SPIFFE credentials. The OAuth authorization server may register additional metadata if needed. The authorization server may use a number of mechanissms for obtaining additional metadata. Metadata may be preconfigured, automatically created, retrieved from a configuration management system, or retrieved from the JWT-SVID, if it was included as additional claims, similar to how Dynamic Client Registration ({{RFC7591}}) describes how a software statement may be used to convey metadata.
```

Comment:
Suggest change to:

The OAuth authorization server avoids operational overheads of registering the client and client secrets by accepting SPIFFE credentials from a workload acting as an OAuth client.